tests/vanilla/store.test.tsx

```diff
@@ -592,3 +592,43 @@ it('should update derived atom even if dependances changed (#2697)', () => {
   store.set(primitiveAtom, 1)
   expect(onChangeDerived).toHaveBeenCalledTimes(1)
 })
+
+it('should preserve dependencies when reusing continuable promise', async () => {
+  const baseAtom = atom(0)
+  let promise = Promise.resolve(-1)
+  let inProgress = false
+  const derivedAtom = atom<Promise<number>, [], void>(
+    (get, { setSelf }) => {
+      if (inProgress) {
+        return promise
```

Comment:
Does this change the behavior?
```suggestion
        return Promise.resolve(-1)
```

Reply:
`return Promise.resolve(store.get(baseAtom))` does not change behavior.
Do you think it should?

Reply:
Ah, it must be the identically same promise in 2.8.4. So the logic needs to be updated to see if the promise is exactly the same.

Reply:
I misunderstood something. Forget about it.